nifi-pulsar-processors/src/test/java/org/apache/nifi/processors/pulsar/pubsub/async/TestAsyncConsumePulsarRecord.java

```diff
@@ -91,36 +104,104 @@ public void singleMessageMultiRecordsTest() throws PulsarClientException {
         assertEquals(expected.toString(), flowFileContents);
     }
 
+    /*
+     * Send multiple messages on different topics,
+     * check if it creates two flow files by retaining the order of messages
+     */
+    @Test
+    public void multipleGoodMessagesOnTwoTopicsCreatesMultipleRecordsTest() throws IOException {
+
+        List<Message<GenericRecord>> mockMessages = new ArrayList<>();
+        mockMessages.add(createTestMessage("A,9".getBytes(), null, singletonMap("prop", "val"), DEFAULT_TOPIC, "", SchemaType.JSON));
```

Comment:
this is a bit disturbing - you are sending in random binary stating it's json and it does not blow up.

Reply:
everything is mocked in this context ... there is no validation to check the schema types.